actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -438,7 +454,20 @@ def form_authenticity_param # :doc:
 
       # Checks if the controller allows forgery protection.
       def protect_against_forgery? # :doc:
-        allow_forgery_protection
+        allow_forgery_protection && ensure_session_is_enabled!
+      end
+
+      def ensure_session_is_enabled!
+        if !session.respond_to?(:enabled?) || session.enabled?
+          true
+        else
+          if Base.silence_disabled_session_errors
```

Comment:
Just one more - this doesn't need to use `Base` as there's an instance accessor.

Reply:
Done.